bwapi/bwresources.py

```diff
@@ -50,34 +50,76 @@ def reload(self):
         if "results" not in response:
             raise KeyError("Could not retrieve" + self.resource_type, response)
 
-        self.ids = {
-            resource["name"]: resource["id"] for resource in response["results"]
+        self.names = {
+            resource["id"]: resource["name"] for resource in response["results"]
         }
 
+    def get_resource_id(self, resource=None, check=False):
+        '''
+        args:
+            check: if set to False (default), this method returns the ID of the resource. 
+            	   if set to True, this method will return True or False, depending on whether the resource exists
+            
+        '''
+        if not resource:
+            if check == False:
+                return "" # return empty string rather than none to avoid stringified "None" becoming part of the url of an API call
+            if check == True:
+                raise ValueError('Please supply a resource to check')
+        if isinstance(resource, int):
+            if resource not in self.names.keys():
+                if check == False:
+                    raise ValueError('Could not find the resource ID {} in the project'.format(resource))
+                elif check == True:
+                    return False
+            pk = resource 
+        elif isinstance(resource, str):
+            entries = [pk for pk, name in self.names.items() if name == resource]
+            if len(entries) > 1:
+                #check status doesn't matter here - it's an error either way
+                raise ValueError('The resource name {} is ambiguous: {}'.format(resource, entries))
+            if entries:
+                if check == False:
+                    return entries[0]
+                elif check == True:
+                    return True
+            else:
+                try: 
+                    pk = int(resource)
+                except:
+                    if check == False:
+                        raise ValueError('Could not find the resource name {} in the project'.format(resource))
+                    elif check == True:
+                        return False
+        if pk not in self.names.keys():
+            if check == False:
+                raise ValueError('Could not find the resource ID {} in the project'.format(resource))
+            elif check == True:
+                return False
+        if pk:
+            if check == False:
+                return pk
+            elif check == True:
+                return True
```

Comment:
Further to my previous comment about "check" I think you could dramatically simplify this method if it only did the `check == False` behaviour. Then you can implement the other method has something like:
```python
def resource_exists(name):
    try:
        get_resource_id(name)
        return True
    except ValueError:
        return False
```